### PR TITLE
 Kcore, EdgeBetweennessCentrality, EdgeNumberDistance,  Categoricalpersistance, EigenVectorCentrality, KCorenesscentrality

### PR DESCRIPTION
**Pull request by egolol**
- [x] CategoricalPersistence.m
- [x] EdgeBetweennessCentrality.m
- [x] EdgeNumberDistance.m
- [x] EigenvectorCentrality.m
- [x] KCore.m
- [x] KCorenessCentrality.m
- [x] Change Distance

CategoricalPersistance and Participation need Multilayer Community Structure, SmallWorld need graph randomization.

### CODE REVIEW

File: braph2genesis/measures/_EdgeBetweennessCentrality.gen.m [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+
+%%% ¡compatible_graphs!
+GraphBD
+GraphBU
+GraphWD
+GraphWU
+MultiplexBD
+MultiplexBU
+MultiplexWD
+MultiplexWU
```
Comment (annacanal) [inline braph2genesis/measures/_EdgeBetweennessCentrality.gen.m:26]: We are forgetting the multigraphBUD /BUT (also multiplex BUD/t, but it wont work directly for them)

File: braph2genesis/measures/_EdgeBetweennessCentrality.gen.m [hunk 2]
```diff
@@ -0,0 +1,272 @@
...
+
+%% ¡props_update!
+
+%%% ¡prop!
+M (result, cell) is the edge betweenness centrality.
+%%%% ¡calculate!
+g = m.get('G'); % graph from measure class
+A = g.get('A'); % cell with adjacency matrix (for graph) or 2D-cell array (for multigraph, multiplex, etc.)
+N = g.nodenumber();
+L = g.layernumber();
```
Comment (annacanal) [inline braph2genesis/measures/_EdgeBetweennessCentrality.gen.m:36]: N and L arent used later

File: braph2genesis/measures/_KCore.gen.m [hunk 3]
```diff
@@ -0,0 +1,208 @@
+%% ¡header!
+KCore < Measure (m, kcore) is the graph kcore.
```
Comment (annacanal) [inline braph2genesis/measures/_KCore.gen.m:2]: KCore < Measure (m, k-core) is the graph k-core.

File: braph2genesis/measures/_KCore.gen.m [hunk 4]
```diff
@@ -0,0 +1,208 @@
...
+GraphWD
+GraphWU
+MultiplexBD
+MultiplexBU
+MultiplexhWD
+MultiplexWU
+
+%% ¡props!
+%%% ¡prop! 
+KCoreThreshold (parameter, SCALAR) is the kcore threshold
```
Comment (annacanal) [inline braph2genesis/measures/_KCore.gen.m:29]: KCoreThreshold (parameter, SCALAR) is the k-core threshold

File: braph2genesis/measures/_KCore.gen.m [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+%%%% ¡default!
+1
+
+%% ¡props_update!
+
+%%% ¡prop!
+M (result, cell) is the kcore.
```
Comment (annacanal) [inline braph2genesis/measures/_KCore.gen.m:36]: M (result, cell) is the k-core.

File: braph2genesis/measures/_KCorenessCentrality.gen.m [hunk 6]
```diff
@@ -0,0 +1,187 @@
+%% ¡header!
+KCorenessCentrality < Measure (m, kcoreness centrality) is the graph kcoreness centrality.
```
Comment (annacanal) [inline braph2genesis/measures/_KCorenessCentrality.gen.m:2]: KCorenessCentrality < Measure (m, k-coreness centrality) is the graph k-coreness centrality.

File: braph2genesis/measures/_KCorenessCentrality.gen.m [hunk 7]
```diff
@@ -0,0 +1,187 @@
...
+GraphBD
+MultiplexWU
+MultiplexWD
+MultiplexBU
+MultiplexBD
+
+%% ¡props_update!
+
+%%% ¡prop!
+M (result, cell) is the kcoreness centrality.
```
Comment (annacanal) [inline braph2genesis/measures/_KCorenessCentrality.gen.m:30]: M (result, cell) is the k-coreness centrality.

**Review by annacanal — CHANGES_REQUESTED** [review]

File: braph2genesis/measures/_EdgeBetweennessCentrality.gen.m (hunk 1 above)
Comment (egolol) [reply braph2genesis/measures/_EdgeBetweennessCentrality.gen.m:26]: We should later come back to this... so we can change everything one time.

Comment (annacanal) [reply braph2genesis/measures/_EdgeBetweennessCentrality.gen.m:26]: yes, maybe once we have all measures in gv-genesis-2 we add these graphs and check everything

**Review by annacanal — APPROVED** [review]

**Review by giovannivolpe — APPROVED** [review]

**Pull request merged**